### ISSUE TITLE
Single-class YOLOv11 model performs significantly better than multi-class model for ball detection

### ISSUE THREAD

**Issue by manoj-samal** [ultralytics/ultralytics#23168]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I am currently using YOLOv11 added p2head for cricket video analysis and have trained two different models:
Imgsize 1280*1280 for both
	1.	Single-class model
	•	Classes: ball
	•	Trained only to detect the cricket ball
	2.	Multi-class model
	•	Classes: ball, batsman, fielder, umpire, ...
	•	Trained on the same dataset distribution but with multiple object classes

Observed Behavior
	•	The single-class (ball-only) model:
	•	Produces much better ball detection accuracy
	•	Has significantly fewer false positives
	•	Is more stable across frames
	•	The multi-class model:
	•	Detects other classes reasonably well
	•	But ball detection quality degrades
	•	Produces more false positives for the ball, especially near players and cluttered regions

Expected Behavior

I expected the multi-class model to have comparable ball detection performance to the single-class model, given:
	•	Same backbone (YOLOv11)
	•	Same input resolution
	•	Similar training data and augmentations

Questions / Clarifications Needed
	1.	Is this degradation in ball performance expected behavior in YOLOv11 when moving from single-class to multi-class training?
	2.	Does YOLOv11 internally bias optimization toward larger or more frequent classes, affecting small objects like the ball?
	3.	Are there recommended strategies for:
	•	Preserving small-object (ball) performance in multi-class models?
	•	Adjusting loss weights, class weights, or hyperparameters?
	4.	Would a two-model pipeline (ball-only model + multi-class model) be the recommended approach for such use cases?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @manoj-samal, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) for new users where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples and where many of the most common questions may already be answered.

Thanks for the detailed comparison between your single-class and multi-class YOLO11 training setup (including the added P2 head) 🙌 This is an automated response to help us gather the right details up front, and an Ultralytics engineer will also assist soon 🤖👀

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to help us debug it.

If this is a custom training ❓ Question, please provide as much information as possible, including dataset image examples and training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

To help us review your specific “ball degrades in multi-class” case faster, please include:
-Your exact training commands (Python or CLI) for both runs
-Your dataset YAML (classes + paths) and class counts per split
-A couple of representative labeled images where false positives occur (especially cluttered regions)
-Train logs/results (including metrics for the ball class) for both runs
-Your environment details (OS, GPU, `ultralytics` version, PyTorch version)

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Can you post `labels.jpg` files from the run folder of the two trainings?

**Comment by 1RaghavM** [comment]
Could you upload a few examples from your dataset, along with the labels? Also if you could give a little more context as to what your training environment looks like in terms of GPU and your training setup. Thanks.

**Comment by glenn-jocher** [comment]
Yes—small, rare classes like a cricket ball often degrade in multi-class training because the classifier/regressor has to share capacity across classes and the ball becomes both underrepresented and easier to confuse with clutter near people (so this behavior can be “expected” unless the dataset/training is carefully balanced); to diagnose, please upload the two runs’ `labels.jpg`, `labels_correlogram.jpg`, `results.csv`, and `args.yaml` (plus your exact `yolo train ...` command) and a small zip of a few representative images/labels where the multi-class model false-positives, and also paste `yolo checks` output so we can rule out env/version mismatches. For preserving ball performance, the most effective levers are usually data balance/oversampling for the ball class, reducing augmentations that shrink/erase tiny objects, and using small-object tactics like tiling/cropping; the docs section on [model evaluation insights (warmup + tiling)](https://docs.ultralytics.com/guides/model-evaluation-insights/) is a good checklist, and if the ball AP is mission-critical, a two-model pipeline is a perfectly reasonable production approach after you confirm labels/class balance aren’t the root cause.
